zbuf/combiner.go

```diff
@@ -55,9 +56,12 @@ type combiner struct {
 	hol     []*zng.Record
 	done    []bool
 	less    RecordCmpFn
+	mu      sync.Mutex
```

Comment:
A Combiner reads from its done slice during reads, and writes to its done slice during close. That can lead to races in a use case like:
```go
rc, err := NewCombine(...)
defer rc.Close()

err = driver.Compile(...)
err = driver.Run(...)
```
Because `driver.Run` now uses the parallel scanner, the `Read()` on the Combiner will happen in a different goroutine than the calling goroutine, where `Close` is called in the defer. This is called out by the race checker when running unit tests.

Combiner is seemingly the only zbuf.ReadCloser that has this property, and using the mutex here seems like the smallest option, but would be glad to hear other suggestions.